src/core/Microsoft.PowerToys.Settings.UI/Controls/HotkeySettingsControl.xaml.cs

```diff
@@ -135,15 +144,106 @@ private void KeyEventHandler(int key, bool matchValue, int matchValueCode, strin
+                    return false;
+                }
+
+                // Shift was not pressed while entering but it was pressed while leaving the hotkey, therefore simulate a shift key press as the system does not know about shift being pressed in the hotkey.
+                else if (internalSettings.Shift && !_shiftKeyDownOnEntering && !internalSettings.Win && !internalSettings.Alt && !internalSettings.Ctrl)
+                {
+                    // This is to reset the shift key press within the control as it was not used within the control but rather was used to leave the hotkey.
+                    internalSettings.Shift = false;
+
+                    NativeKeyboardHelper.INPUT inputShift = new NativeKeyboardHelper.INPUT
+                    {
+                        type = NativeKeyboardHelper.INPUTTYPE.INPUT_KEYBOARD,
+                        data = new NativeKeyboardHelper.InputUnion
+                        {
+                            ki = new NativeKeyboardHelper.KEYBDINPUT
+                            {
+                                wVk = 0x10,
+                                dwFlags = (uint)NativeKeyboardHelper.KeyEventF.KeyDown,
+
+                                // Any keyevent with the extraInfo set to this value will be ignored by the keyboard hook and sent to the system instead.
+                                dwExtraInfo = (UIntPtr)extraInfoIgnoreKeyEvent,
+                            },
+                        },
+                    };
+
+                    NativeKeyboardHelper.INPUT[] inputs = new NativeKeyboardHelper.INPUT[] { inputShift };
+
+                    _ = NativeMethods.SendInput(1, inputs, NativeKeyboardHelper.INPUT.Size);
+
+                    return false;
+                }
+
+                // Shift was pressed on entering and remained pressed, therefore only ignore the tab key so that it can be passed to the system.
+                // As the shift key is already assumed to be pressed by the system while it entered the hotkey control, shift would still remain pressed, hence ignoring the tab input would simulate a Shift+Tab key press.
+                else if (!internalSettings.Shift && _shiftKeyDownOnEntering && !_shiftToggled && !internalSettings.Win && !internalSettings.Alt && !internalSettings.Ctrl)
+                {
+                    return false;
+                }
+
+                // Shift was pressed on entering but it was released and later pressed again.
+                // Ignore the tab key and the system already has the shift key pressed, therefore this would simulate Shift+Tab.
+                // However, since the last shift key was only used to move out of the control, reset the status of shift within the control.
+                else if (internalSettings.Shift && _shiftKeyDownOnEntering && _shiftToggled && !internalSettings.Win && !internalSettings.Alt && !internalSettings.Ctrl)
+                {
+                    internalSettings.Shift = false;
```

Comment:
[nit] why is this required while leaving the control?

Reply:
If this value is not set to false, then the next time we enter the control it would assume that shift was pressed within the control even though it wasn't and go into that block of if-else code. Pressing tab would then simulate shift-tab. 

As you can see if in the gif, I initially pressed tab and shift+tab and released shift within the control. Then i pressed shift+Tab to exit the control and so the next time i tabbed into the control, since the internalsettings.shift was true it assumed that shift was pressed and simulated a shift+tab instead.

![unexpected_shift_behavior](https://user-images.githubusercontent.com/28739210/95266356-df4ae780-07e7-11eb-8c58-bf81761fd3cd.gif)

Reply:
I feel like this would be fixed if we call `KeyEventHandler` at the start of FilterKeyboardEvent right?

Reply:
I'm not sure I get what you're saying. The keyEventHandler is used to set the internalsettings status, ie. the status of each key according to the control. Therefore whenever there is a key_up or key_down, the internalsettings is updated accordingly using the keyeventhandler. I could call keyEventHandler with shift key up but that's a convoluted way of resetting the internalsettings status of the shift key to false.